### PR TITLE
Add Option for disable observer clocks

### PR DESCRIPTION
**Pull request by Euphillya**
Allow to disable observer clocks: https://www.spigotmc.org/attachments/observerclock-gif.365936/

### CODE REVIEW

File: patches/server/0232-Add-Option-for-disable-observer-clocks.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+     @Override
+     public BlockState updateShape(BlockState state, Direction direction, BlockState neighborState, LevelAccessor world, BlockPos pos, BlockPos neighborPos) {
+-        if (state.getValue(ObserverBlock.FACING) == direction && !(Boolean) state.getValue(ObserverBlock.POWERED)) {
++        // Purpur Start
++        if (state.getValue(ObserverBlock.FACING ) == direction && !(Boolean) state.getValue(ObserverBlock.POWERED)
++            && (!world.getMinecraftWorld().purpurConfig.disableObserverClocks || !(neighborState.getBlock() instanceof ObserverBlock) ||
++                neighborState.getValue(ObserverBlock.FACING).getOpposite() != direction)) {
+             this.startSignal(world, pos);
+         }
++        // Purpur Stop
```
Comment (katekerllenevich) [inline patches/server/0232-Add-Option-for-disable-observer-clocks.patch]: end not stop

**Review by katekerllenevich — CHANGES_REQUESTED** [review]

File: patches/server/0232-Add-Option-for-disable-observer-clocks.patch (hunk 1 above)
Comment (Euphillya) [reply patches/server/0232-Add-Option-for-disable-observer-clocks.patch]: oh my bad sorry ^^

File: patches/server/0232-Add-Option-for-disable-observer-clocks.patch [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
++++ b/src/main/java/net/minecraft/world/level/block/ObserverBlock.java
+@@ -63,9 +63,13 @@ public class ObserverBlock extends DirectionalBlock {
+ 
+     @Override
+     public BlockState updateShape(BlockState state, Direction direction, BlockState neighborState, LevelAccessor world, BlockPos pos, BlockPos neighborPos) {
+-        if (state.getValue(ObserverBlock.FACING) == direction && !(Boolean) state.getValue(ObserverBlock.POWERED)) {
++        // Purpur Start
++        if (state.getValue(ObserverBlock.FACING ) == direction && !(Boolean) state.getValue(ObserverBlock.POWERED)
++            && (!world.getMinecraftWorld().purpurConfig.disableObserverClocks || !(neighborState.getBlock() instanceof ObserverBlock) ||
++                neighborState.getValue(ObserverBlock.FACING).getOpposite() != direction)) {
```
Comment (katekerllenevich) [inline patches/server/0232-Add-Option-for-disable-observer-clocks.patch]: while putting parts of the if statement on a new line help for readability, it makes updating harder. please keep if statements on the same line ^_^

**Review by katekerllenevich — CHANGES_REQUESTED** [review]

File: patches/server/0232-Add-Option-for-disable-observer-clocks.patch (hunk 2 above)
Comment (Euphillya) [reply patches/server/0232-Add-Option-for-disable-observer-clocks.patch]: It is modified, and indeed that will make it possible to modify it better. I just thought it was a very long line on my IDE.

**Review by katekerllenevich — APPROVED** [review]

**Comment by Euphillya** [comment]
up

**Pull request merged**